src/components/registerPlugins.ts

```diff
@@ -0,0 +1,105 @@
+// Runs at server start. It checks the (Action) plugins folder
+// and the Database action table and loads any new plugins into the database
+// It should also give a warning about any plugins in the database that are
+// no longer available in the system and removes them from the Database.
+// Should also compare and update name, description, etc.
+// The unique identifier for a plugin is its "code".
+
+import * as fs from 'fs';
+import path from 'path';
+import { DatabaseResult } from './triggersAndActions';
+import getAppRootDir from './getAppRoot';
+
+const pluginFolder = path.join(getAppRootDir(), 'plugins'); //Change this to config file
+
+export default async function registerPlugins(client: { [key: string]: Function }) {
+  // Load plugin info from files
+  console.log('Scanning plugins folder...');
+  const plugins = fs
+    .readdirSync(pluginFolder)
+    .filter((item) => fs.statSync(path.join(pluginFolder, item)).isDirectory())
+    .map((item) => {
+      return { path: path.join(item, 'src') };
+    })
+    .filter((item) => fs.existsSync(path.join(pluginFolder, item.path, 'plugin.json')))
+    .map((item) => {
+      const pluginObject = {
+        ...item,
+        ...JSON.parse(fs.readFileSync(path.join(pluginFolder, item.path, 'plugin.json'), 'utf8')),
+      };
```

Comment:
It won't because the previous `.filter` removes any items from the array that don't have the .`json` file at the expected path. ;)

(For testing, I originally had additional dummy plugin folders that were missing `plugin.json` files, and they were skipped gracefully).

Reply:
Great! That's nice touch. It would be good to add to the documentation (if you haven't already) what what is expected to have inside each plugin folder.

Reply:
Okay have made [an issue](#21) for it. :)